packages/react-components/src/organisms/CheckboxGroup.tsx

```diff
@@ -25,16 +26,24 @@ export default function CheckboxGroup<V extends string>({
   const groupName = useRef(uuidV4());
   return (
     <>
-      {options.map((option, index) => (
-        <LabeledCheckbox
-          key={`${groupName}-${index}`}
-          groupName={groupName.current}
-          title={option.label}
-          enabled={option.enabled}
-          checked={values.has(option.value)}
-          onSelect={() => onChange(option.value)}
-        />
-      ))}
+      {options.map((option, index) =>
+        option.value === undefined ? (
+          <Caption asParagraph>
+            <strong>{option.label}</strong>
+          </Caption>
+        ) : (
+          <LabeledCheckbox
+            key={`${groupName}-${index}`}
+            groupName={groupName.current}
+            title={option.label}
+            enabled={option.enabled}
+            checked={values.has(option.value)}
+            onSelect={() =>
+              option.value !== undefined && onChange(option.value)
```

Comment:
I don't think you need the undefined check you have already done it above
```suggestion
onChange(option.value)
```